### PR TITLE
DB-7692: stop vacuum during database restore

### PR DESCRIPTION
**Pull request by jyuanca**
Please review code changes on ee branch as well.

### CODE REVIEW

File: splice_machine/src/main/java/com/splicemachine/derby/utils/Vacuum.java [hunk 1]
```diff
@@ -199,6 +199,9 @@ public void vacuumDatabase(long oldestActiveTransaction) throws SQLException{
 
             List<DataDelete> deletes = new ArrayList<>();
             for (long removed : toDelete) {
+                boolean restoreMode = SIDriver.driver().lifecycleManager().isRestoreMode();
+                if (restoreMode)
+                    break;
```
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/derby/utils/Vacuum.java]: Should we break out here or return directly? It is possible that when restoreMode becomes true, the _deletes_ list is non-empty, in this case, do we want "dropped.delete(deletes)" to be executed or not?

Comment (jyuanca) [reply splice_machine/src/main/java/com/splicemachine/derby/utils/Vacuum.java]: Good point. We may want to terminate vacuum immediately during a database restore.

File: splice_machine/src/main/java/com/splicemachine/derby/utils/Vacuum.java [hunk 2]
```diff
@@ -199,6 +199,9 @@ public void vacuumDatabase(long oldestActiveTransaction) throws SQLException{
 
             List<DataDelete> deletes = new ArrayList<>();
             for (long removed : toDelete) {
+                boolean restoreMode = SIDriver.driver().lifecycleManager().isRestoreMode();
+                if (restoreMode)
+                    return;
```
Comment (yxia92) [inline splice_machine/src/main/java/com/splicemachine/derby/utils/Vacuum.java]: On a second thought, I think this code may have an issue. When we get here, all the eligible dropped tables have been deleted, and this FOR loop is just to prepare for the deletion of the corresponding entries in the DROPPED_CONGLOMERATES_TABLE. If we abort here, the tables deleted will never get a chance to be removed from the DROPPED_CONGLOMERATES_TABLE. Correct me if I missed something.

Comment (dgomezferro) [reply splice_machine/src/main/java/com/splicemachine/derby/utils/Vacuum.java]: That's right @yxia92 , we want to break out of the "for(TableDescriptor table:hTableDescriptors){" loop, not this one.

**Comment by jyuanca** [comment]
@yxia92 @dgomezferro I think that's my original intention, but I later got confused ...

**Pull request merged**